android-components/components/support/utils/src/main/java/mozilla/components/support/utils/WebURLFinder.kt

```diff
@@ -96,29 +96,27 @@ class WebURLFinder {
         }
 
         /**
-         * Check if string is a Web URL.
+         * Checks if the given [String] is a valid web URL.
          *
+         * Valid URI schemes: 'http:', 'https:', 'about:', 'data:'.
          *
-         * A Web URL is a URI that is not a `file:` or
-         * `javascript:` scheme.
+         * Invalid URI schemes: 'file:', 'javascript:', 'content:'.
          *
-         * @param string to check.
-         * @return `true` if `string` is a Web URL.
+         * @return True if the [String] is a valid web URL.
          */
         @SuppressWarnings("TooGenericExceptionCaught")
-        fun isWebURL(string: String): Boolean {
-            try {
-                URI(string)
-            } catch (e: Exception) {
-                return false
-            }
+        fun String.isValidWebURL() = try {
+            URI(this)
 
-            return !(
-                URLUtil.isFileUrl(string.lowercase(Locale.ROOT)) ||
-                    URLUtil.isJavaScriptUrl(string.lowercase(Locale.ROOT))
-                )
+            val safeUri = lowercase(Locale.ROOT)
```

Comment:
this is kinda subtle since it's an extension function, but wondering aloud (non blocking) if it'd be clearer as `this.lowercase`

Reply:
Fair question, I would say I prefer omitting the `this` as I find it a bit noisy, but I can see your thinking. It perhaps looks more subtle in the review format as opposed to the IDE?